lib/runners/processor/metrics_fileinfo.rb

```diff
@@ -97,6 +109,56 @@ def analyze_last_committed_at(targets)
       end
     end
 
+    def code_churn
+      @code_churn ||= {}
+    end
+
+    def analyze_code_churn
+      trace_writer.message "Analyzing code churn..." do
+        count_by_num, commits_by_num = commits_within("-n", "100")
+        days_ago = (DateTime.parse(commits_by_num[:latest][:datetime]) - 90).iso8601
+        count_by_time, commits_by_time = commits_within("--since", days_ago)
+        outlive_commits = count_by_num > count_by_time ? commits_by_num : commits_by_time
+
+        stdout, _ = capture3!("git", "log", "--reverse", "--format=format:%cI", "--numstat", "#{outlive_commits[:oldest][:sha]}..HEAD", **CAP3ARGS_SUPPRESS_TRACE)
+        lines = stdout.lines(chomp: true)
+        @number_of_commits = 0
+        lines.reject(&:empty?).each do |line|
+          if line.include?("\t") then
+            parse_numstat_line(line)
```

Comment:
[suggestion] How about the following refactoring to avoid the duplicated use of the `\t` separator?

```suggestion
          adds, dels, fname = line.split("\t")
          if adds && dels && fname
            parse_numstat_line(adds, dels, fname)
```

Reply:
Then, `calc_churn` seems better than the method name `parse_numstat_line`.

Reply:
The `calc_churn` method will be simpler as below:

```ruby
code_churn[fname] = calc_churn(code_churn[fname], adds, dels)

# ...

def calc_chrun(churn, adds, dels)
  churn ||= { occurrence: 0, additions: 0, deletions: 0 }
  churn[:occurrence] += 1
  churn[:additions] += adds == "-" ? 0 : Integer(adds)
  churn[:deletions] += dels == "-" ? 0 : Integer(dels)
  churn
end
```

Reply:
Good catch! Fixed. commit 74a905f6b697d5de894cfa3f206b816a0c848eb8